File: Rails4/app/models/post.rb
class Post < ActiveRecord::Base
  extend FriendlyId
  friendly_id :name, use: :slugged

  belongs_to :category
  belongs_to :user
  has_many :comments

  validates :name, presence: true
  validates :content, presence: true
  validates :category, presence: true
  validates :user, presence: true

  after_create :set_post_count
  after_destroy :set_post_count

  def set_post_count
    category.post_count = category.posts.count
    category.save
  end
end

File: Laravel4/app/models/Post.php
<?php

class Post extends Eloquent {
	protected $guarded = array();

	private $rules = array(
		'name' => 'required',
		'slug' => 'required|unique:posts',
		'category_id' => 'exists:categories,id',
		'user_id' => 'exists:users,id',
		'content' => 'required',
	);

	public static $sluggable = array(
		'build_from' => 'name',
		'save_to'    => 'slug',
	);

	public function category()
	{
		return $this->belongsTo('Category');
	}

	public function author()
	{
		return $this->belongsTo('User', 'user_id');
	}

	public function comments()
	{
		return $this->hasMany('Comment');
	}

	public function validate($data, $rules = array())
	{
		$rules = array_merge($this->rules, $rules);

		$v = Validator::make($data, $rules);

		if ($v->fails())
		{
			$this->errors = $v->messages();
			return false;
		}

		return $v->passes();
	}

	public function errors()
	{
		return $this->errors;
	}
}
